src/CRC32.cpp

```diff
@@ -32,7 +32,7 @@ void CRC_Init()
 	}
 }
 
-u32 CRC_Calculate( u32 crc, const void * buffer, u32 count )
+u64 CRC_Calculate( u64 crc, const void * buffer, u32 count )
 {
 	u8 *p;
 	u32 orig = crc;
```

Comment:
It looks suspicious now.
I suppose, it would be safer to rewrite it as:
```
        u32 crc32 = static_cast<u32>(crc);
        u32 orig = crc32;
	p = (u8*) buffer;
	while (count--)
		crc32 = (crc32 >> 8) ^ CRCTable[(crc32 & 0xFF) ^ *p++];
	return crc32 ^ orig;
}
```

The same is for CRC_CalculatePalette bellow and for functions from CRC32_ARMV8.cpp